Add typed signing

bonustrack opened the pull request.

samuveth commented:
I have this issue when trying to login with Fortmatic 
![image](https://user-images.githubusercontent.com/51686767/124536615-765dd200-de42-11eb-82af-a09ea5534a4f.png)

Creating a proposal with Portis didn't work for me
![image](https://user-images.githubusercontent.com/51686767/124536831-ed936600-de42-11eb-9961-29e60b249d89.png)

bonustrack commented:
@samuveth Thanks, i've updated Portis module that seem to have fixed the issue. Fortmatic issue is also on snapshot.org too, we can find a fix on another PR, i've removed it for now.

samuveth commented:
@bonustrack Hm, I'm still getting an error when trying to create a proposal with Portis.
![image](https://user-images.githubusercontent.com/51686767/124715096-c87b2200-df2c-11eb-8873-ff456b865e9a.png)

ChaituVR commented:
@samuveth same with trust's walletconnect..  looks like `"type": "string[]"` this kind of structure is not supported in many wallets

samuveth commented:
With Argent and Wallet Connect I'm getting this error 
![image](https://user-images.githubusercontent.com/51686767/124715629-67078300-df2d-11eb-97c1-5ce60ea259ab.png)

bonustrack commented:
@samuveth Argent on iPhone or Android?

samuveth commented:
> @samuveth Argent on iPhone or Android?

Android

bonustrack commented:
@samuveth For the Argent error, do you get it for create a proposal or vote?

samuveth commented:
> @samuveth For the Argent error, do you get it for create a proposal or vote?

Create proposal, I haven't tested voting yet

samuveth commented:
- after proposal creation `getProposal` returns `null`
- when trying to vote I get
![image](https://user-images.githubusercontent.com/51686767/128869670-aba072e0-2030-4960-be42-a65c561c83ab.png)
- when trying to delete a proposal I get
![image](https://user-images.githubusercontent.com/51686767/128869829-9d4d7310-28e6-4372-a177-099e9fbc9942.png)
- when trying to save settings I get
![image](https://user-images.githubusercontent.com/51686767/128870011-fff0a57e-64e0-4764-a2bf-694ce1283808.png)

bonustrack commented:
@samuveth Cant reproduce the error you got, tested a vote, proposal, delete and settings all works fine for me. Can you reproduce in the deployment here? https://bafybeid7uwlz7n3sytfgrsp3og6hqz4ozmqf6uwiz3jdlww2jakx7lg42u.on.fleek.co

samuveth commented:
> @samuveth Cant reproduce the error you got, tested a vote, proposal, delete and settings all works fine for me. Can you reproduce in the deployment here? https://bafybeid7uwlz7n3sytfgrsp3og6hqz4ozmqf6uwiz3jdlww2jakx7lg42u.on.fleek.co

@bonustrack on this build it's fine, but locally it doesn't work for me even with a fresh clone of the repo. Have you tried locally?

bonustrack commented:
@samuveth are you sure that you use the latest version of Snapshot.js?

bonustrack commented:
@samuveth It should be fixed, there was a "process.env." in the code instead of "import.meta.env.", it failed because your tried to vote on a proposal from hub.snapshot.org but the message was sent to testnet.snapshot.org

bonustrack commented:
Can you fix conflicts @samuveth ?

samuveth commented:
@bonustrack I think a might have done something wrong with the `yarm.lock` on the merge. Can you see something wrong?

bonustrack commented:
@samuveth There shouldn't be any change on yarn.lock with this PR

bonustrack commented:
Something is weird, when i try push yarn.lock from develop branch in this PR it still show change on GitHub on the file.

samuveth commented:
@bonustrack Did you try to revert the merge I did?

bonustrack commented:
@samuveth Not really, i've just tried to fix conflicts

samuveth commented:
Some testing that I've done so far:
- Metamask inject: everything works
- WalletConnect with Metamask mobile: everything works except creating a proposal, no error but seems to be stopping at `signer._signTypedData()`
- WalletConnect with Argent: everything works except creating a proposal, error `wrong signature`

To be continued
